### PR TITLE
Fixed examples and datagen so that noiseless filter runs have zero error

### PR DESCRIPTION
**Pull request by CharlesCossette**
we should be able to get those integration tests working now

### CODE REVIEW

File: examples/ex_ekf_vector.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
     range_freqs,
 )
 
-gt_data, input_data, meas_data = dg.generate(x0, 0, 10, noise=True)
+gt_data, input_data, meas_data = dg.generate(x0, 0, 10, noise=noise_active)
 
 # ##############################################################################
 # Run Filter
```
Comment (vkorotkine) [inline examples/ex_ekf_vector.py:50]: Should x be corrupted with random sample drawn from P0? And be exectued if noise_active = True

File: examples/ex_invariant_so3.py [hunk 2]
```diff
@@ -31,9 +38,9 @@
     [mag_model, grav_model],
     1,
 )
-state_true, input_list, meas_list = dg.generate(x0, 0, 30, True)
-
+state_true, input_list, meas_list = dg.generate(x0, 0, 30, noise_active)
 
+# ##############################################################################
 # Run the regular filter
 x0.direction = "right"
```
Comment (vkorotkine) [inline examples/ex_invariant_so3.py:45]: Same here for x0, should be corrupted with P0 if noise_active

File: examples/ex_iterated_ekf_se2.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def input_profile(t, x):
 # ##############################################################################
 # Data Generation
 dg = DataGenerator(process_model, input_profile, Q, 200, range_models, 10)
-state_true, input_data, meas_data = dg.generate(x0, 0, 10, noise=True)
+state_true, input_data, meas_data = dg.generate(x0, 0, 10, noise=noise_active)
 
 # %% ###########################################################################
 # Run Filter
```
Comment (vkorotkine) [inline examples/ex_iterated_ekf_se2.py:43]: Same here for x0, should be corrupted with P0 if noise_active

File: examples/ex_ekf_vector.py (hunk 1 above)
Comment (CharlesCossette) [reply examples/ex_ekf_vector.py:50]: yup you're right. i'll make those fixes!

File: examples/ex_invariant_so3.py [hunk 4]
```diff
@@ -31,9 +38,9 @@
     [mag_model, grav_model],
     1,
 )
-state_true, input_list, meas_list = dg.generate(x0, 0, 30, True)
-
-
+state_true, input_list, meas_list = dg.generate(x0, 0, 30, noise_active)
+x0.plus(randvec(P0))
```
Comment (vkorotkine) [inline examples/ex_invariant_so3.py:42]: If noise_active == False, no initial condition shouldnt be corrupted right?
```suggestion
if noise_active:
    x0.plus(randvec(P0))
```

**Pull request merged**